node/network/statement-distribution/src/lib.rs

```diff
@@ -1341,8 +1346,17 @@ async fn handle_incoming_message_and_circulate<'a>(
 		// that require dependents. Thus, if this is a `Seconded` statement for a candidate we
 		// were not aware of before, we cannot have any dependent statements from the candidate.
 		let _ = metrics.time_network_bridge_update_v1("circulate_statement");
+
+		let session_index = runtime
+			.get_session_index_for_child(ctx.sender(), relay_parent)
+			.await
+			.unwrap_or(0_u32);
```

Comment:
Nit: log `get_session_index_for_child()`. Not sure how often this happens, but I guess `TRACE` is fine.

Are there any issues with using the topology from ancient `session 0` ?

Reply:
I just saw that the `get_topology` fn will return the current topology in this case. Please see my other comment.